apollo-compiler/src/test/kotlin/com/apollographql/apollo/compiler/CodegenTest.kt

```diff
@@ -218,16 +175,21 @@ class CodeGenTest(private val folder: File) {
           generateVisitorForPolymorphicDatatypes = generateVisitorForPolymorphicDatatypes,
           generateAsInternal = generateAsInternal,
           kotlinMultiPlatformProject = true,
-          enumAsSealedClassPatternFilters = enumAsSealedClassPatternFilters
+          enumAsSealedClassPatternFilters = enumAsSealedClassPatternFilters,
+          metadataOutputFile = File("build/generated/test/${folder.name}/metadata/$language"),
       )
     }
 
     @JvmStatic
-    @Parameterized.Parameters(name = "{0}")
-    fun data(): Collection<File> {
+    @Parameterized.Parameters(name = "{0}-{1}")
+    fun data(): Collection<Array<Any>> {
       return File("src/test/graphql/com/example/")
           .listFiles()!!
           .filter { it.isDirectory }
+          .flatMap { listOf(
+              arrayOf(it, TestLanguage.Java),
+              arrayOf(it, TestLanguage.Kotlin)
+          ) }
```

Comment:
Refactored these tests slightly to have one test for Java and another for Kotlin